### DISCUSSION TITLE
Dynamically change X-Hasura-Role

### DISCUSSION THREAD

**Discussion by techpet** [nhost/nhost#1099] [General]
Is it possible to dynamically set X-Hasura-Role in the JWT after a user is logged in. Does one have to manually implement a function that creates a new JWT with the desired role after user performed some actions and then sets this JWT as the new access token in order to achieve such functionality?

**Comment by plmercereau** [comment]
`x-hasura-role` doesn't need to be added to the JWT, but sent as an additional header together with the `Authorization` header.
You can give it any value from the `x-hasura-allowed-roles` (that is encoded in the JWT). If none is given, it will fall back to the `x-hasura-default-role` value (also in the JWT).

You can have a look at the [Hasura documentation](https://hasura.io/docs/latest/auth/authentication/jwt/), that explains it much better than I do.

**Reply by techpet** [reply]
Thank you @plmercereau ! Don't know how I overlooked this in Hasura documentation but that was exactly what I wanted!
